flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/enumerator/KafkaSourceEnumerator.java

```diff
@@ -131,7 +131,14 @@ public void start() {
 			LOG.info("Starting the KafkaSourceEnumerator for consumer group {} " +
 				"without periodic partition discovery.", consumerGroupId);
 			context.callAsync(
-					this::discoverAndInitializePartitionSplit,
+					() -> {
+						try {
+							return discoverAndInitializePartitionSplit();
+						} finally {
+							// Close the admin client early because we won't use it anymore.
+							adminClient.close();
```

Comment:
Is the admin client created in the `start()` method and closed in the start method? Should it be a local variable then? Having it as a class field suggest other methods can use it.

Edit: Seems it is used in other places, which seems a bit hard to understand now. What is the life-cycle and use of that field?